wild_lib/src/args.rs

```diff
@@ -252,7 +252,13 @@ pub(crate) fn parse<S: AsRef<str>, I: Iterator<Item = S>>(mut input: I) -> Resul
                 bail!("Unsupported hash-style `{style}`");
             }
             // Since we currently only support GNU hash, there's no state to update.
-        } else if long_arg_split_prefix("build-id=").is_some() {
+        } else if long_arg_eq("build-id") {
+            build_id = true;
+        } else if let Some(build_id_value) = long_arg_split_prefix("build-id=") {
```

Comment:
Consider also supporting a user-defined set hash, similar to other linkers do:

```
$ ld.lld --help | less
  --build-id=[fast,md5,sha1,uuid,0x<hexstring>]
                          Generate build ID note
  --build-id              Alias for --build-id=sha1
```

Reply:
Just to confirm, are you requesting support for a --build-id=0x<hex> style option where users can specify the Build ID explicitly?

Reply:
Yes, please. For all the other option values (`fast`, `md5`, `sha1`) except `uuid`, I would use the BLAKE3 hashing algorithm.